### ISSUE TITLE
Test accuracy on Structured3D much lower than paper

### ISSUE THREAD

**Issue by yuanchen798** [felix-ch/f3loc#13]
[data_utils.py](https://github.com/user-attachments/files/22042181/data_utils.py)
[demo-s3d.py](https://github.com/user-attachments/files/22042049/demo-s3d.py)

Hi, I'm trying to reproduce F3Loc's test on the Structured3D dataset. But my recall is much lower than what's written in the paper. I saw that the depth prediction by the monocular network is far from the ground truth. I used the perspective-full data and implement the mask for gravity-aligned images. The camera intrinsics in gravity alignment is set the same as provided in S3D data_utils.py.

Should I train the model first on the S3D trainig set and then turn to test set? Is there anything else I should pay attention to?

Thank you in advance.

**Comment by felix-ch** [comment]
Hi,
Yes, the provided checkpoints are for the gibson floorplan dataset. To test on S3D perspective full, you need to train on its training set first. The model is not meant to generalize across different cameras or domains.
Best

**Comment by yuanchen798** [comment]
OK thanks. Is it trained from a brand new model or from the existed checkpoint? If I should start with a brand new one, any recommendation about initialization?

**Comment by yuanchen798** [comment]
Or would it possible to provide the ckpt for S3D? That would make things much easier.

**Comment by felix-ch** [comment]
The model is initialized in the same way for all dataset as in the code. Resnet pretrained and others from scratch. Sorry we cannot provide the ckpt for S3D.
